quickjs.c

```diff
@@ -1812,6 +1814,9 @@ int JS_EnqueueJob(JSContext *ctx, JSJobFunc *job_func,
     JSJobEntry *e;
     int i;
 
+    if (rt->in_free)
+        return -1;
```

Comment:
This is going to result in leaking the argv if the caller doesn't check the return code. Which we don't in many places, and where we do, we expect an exception to have been raised.

Reply:
Not really because enqueue dups the arguments, so the caller doesn't dup them.

Reply:
You could indeed argue there is no exception raised, but since this happens in such a weird spot, I thought I'd just get away with it :-P

Reply:
Alternatively, I could check if `rt->in_free` when resetting the weakref, and just avoid the whole thing, but calling JS_EnqueueJob when the runtime is going away still leads to leaks.

Reply:
I'd probably scope it narrowly to reset_weak_ref() and maybe add an assert to JS_EnqueueJob() because it's probably always a bug if you call that when tearing down the runtime.

Reply:
Can do!

Reply:
Updated!